File: Solution/Anmol_Chawla_task1.scala
import org.apache.spark.{SparkConf, SparkContext}

object task1 {
  def main(args: Array[String]) {
    val conf = new SparkConf().setMaster("local").setAppName("task1").setSparkHome("src/main/resources")
    val sc = new SparkContext(conf)
    val spark = org.apache.spark.sql.SparkSession.builder.master("local").appName("task1").getOrCreate;
    val df = spark.read.format("csv").option("header", "true").option("mode", "DROPMALFORMED").option("inferSchema","true").load(args(0))
    val df1 = df.drop("userId", "timestamp").groupBy("movieId").avg("rating").sort("movieId").withColumnRenamed("avg(rating)", "rating_avg").coalesce(1).write.format("csv").option("header","true").save(args(1))
  }
}

File: Solution/Anmol_Chawla_task1.py
import sys
from pyspark import SparkContext
from pyspark import SparkConf
from pyspark.sql import SQLContext
from pyspark.sql import Row
from pyspark.sql.functions import *

sc = SparkContext.getOrCreate()
sqlContext = SQLContext(sc)


ip_csv_file = sys.argv[1]
op_csv_file = sys.argv[2]


df = sqlContext.read.options(header='true', inferschema='true').csv(ip_csv_file)
data = df.drop("userId","timestamp").sort('movieId').groupby('movieId').avg('rating').select(col("movieId").alias("movieId"),col("avg(rating)").alias("rating_avg")).coalesce(1).write.options(header="true").csv(op_csv_file)